tools/ms_register.sh

```diff
@@ -0,0 +1,184 @@
+	cp -rf template/$1 .
+	mv $1 $2
+	CLASS_PATH="$PWD"/"$2"/class.txt
+	cd $2
+	find ./ -type f -exec sed -i "s/template/$2/g" {} \;
+	find ./ -type f -exec sed -i "s/TPL/$3/g" {} \;
+	find . -depth -name '*template*' -execdir rename "s/template/$2/g" {} \;
+	find . -depth -name '*TPL*' -execdir rename "s/TPL/$3/g" {} \;
+	cd ..
+	echo "[Info] Template folder is ready!"
+}
+
+OP=""
+if [ "$1" = "add" ]; then
+	OP="add"
+elif [ "$1" = "delete" ]; then
+	OP="delete"
+else
+	echo "### Specify what you want to do (add or delete)"
+	printf "### Enter you want to do: "
+	read OP
+fi
+
+if [ "$OP" = "add" ]; then
+	NAME_VALID=1
+	while [ $NAME_VALID -ne 0 ]; do
+		echo "### Specify your service name (e.g. musicservice)."
+		printf "### Enter your service name: "
+		read NAME
+		if [ "$NAME" = "" ]; then
+			echo "[Error] Service name cannot be empty! Please try another one!"
+		else
+			check_valid NAME $NAME
+			NAME_VALID=$?
+			if [ $NAME_VALID -ne 0 ]; then
+				echo "[Error] Service name used! Please try another one!"
+			fi
+		fi
+	done
+
+	echo ""
+	ACN_VALID=1
+	while [ $ACN_VALID -ne 0 ]; do
+		echo "### Specify the acronym of your service (e.g. MS)."
+		printf "### Enter the acronym of your service: "
+		read ACN
+		if [ "$ACN" = "" ]; then
+			echo "[Error] Service acronym cannot be empty! Please try another one!"
+		else
+			check_valid ACN $ACN
+			ACN_VALID=$?
+			if [ $ACN_VALID -ne 0 ]; then
+				echo "[Error] Service acronym used! Please try another one!"
+			fi
+		fi
+	done
+
+	echo ""
+	echo "### Specify the programming language you want to you in your programming. If C++/Java/Python, then template will be provided."
+	printf "### Enter the programming language: "
+	read LAN
+
+	echo ""
+	HOST_PORT_VALID=1
+	while [ $HOST_PORT_VALID -ne 0 ]; do
+		HOST_VALID=1
+		while [ $HOST_VALID -ne 0 ]; do
+			echo "### Specify the host/port information for your service. "
+			printf "### Enter the host: "
+			read HOST
+			if [ "$HOST" = "" ]; then
+				echo "[Error] Service host cannot be empty! Please try another one!"
+			else
+				HOST_VALID=0
+			fi
+		done
+
+		PORT_VALID=1
+		while [ $PORT_VALID -ne 0 ]; do
+			printf "### Enter the port: "
+			read PORT
+			if [ "$PORT" = "" ]; then
+				echo "[Error] Service port cannot be empty! Please try another one!"
+			else
+				PORT_VALID=0
+			fi
+		done
+		
+		check_valid HOST_PORT $HOST $PORT
+		HOST_PORT_VALID=$?
+		if [ $HOST_PORT_VALID -ne 0 ]; then
+			echo "[Error] Service host/port pair already used! Please try another one!"
+		fi
+	done
+
+	echo ""
+	INPUT_VALID=1
+	while [ $INPUT_VALID -ne -0 ]; do
+		echo "Specify the input type of your service (text, image or text_image)"
+		printf "### Enter the input type: "
+		read INPUT
+		if [ "$INPUT" = "" ]; then
+			echo "[Error] Service input type cannot be empty! Please try another one!"
+		else
+			INPUT_VALID=0
+		fi
+	done
+
+	if [ "$LAN" = "cpp" -o "$LAN" = "java" -o "$LAN" = "python" ]; then
+		# do copy template folder of cpp to lucida
```

Comment:
This naming needs to be case insensitive. If I type "Java", it fails to give me a template. Also should include "c++" as an entry instead of only "cpp" (with allowing for case insensitivity).